packages/flutter/lib/src/material/flat_button.dart

```diff
@@ -40,6 +40,44 @@ import 'theme_data.dart';
 ///
 /// The [clipBehavior] argument must not be null.
 ///
+/// {@tool sample}
+///
+/// This example shows how to use [FlatButton] to create a simple button
+/// in your flutter code.
+///
+/// ```dart
+/// FlatButton(
+///   onPressed: () {
+///     /*...*/
+///   },
+///   child: Text(
+///     "Flat Button",
+///     style: TextStyle(fontSize: 20.0),
+///   ),
+/// )
+/// ```
+/// This example shows how to use [FlatButton] to create a styled
```

Comment:
Each example should probably be wrapped in a seperate `{@tool sample} ... {@end-tool}` clause.